work/builder/builder.go

```diff
@@ -37,6 +38,16 @@ func buildDependencyIndices(txs []*TxOrGen, bundles []*Bundle, signer types.Sign
 	senderToIndices := make(map[common.Address][]int)
 	bundleToIndices := make(map[int][]int)
 
+	txHashToBundleIndices := make(map[common.Hash][]int)
+	for i, bundle := range bundles {
+		for _, tx := range bundle.BundleTxs {
+			txHashToBundleIndices[tx.Id] = append(txHashToBundleIndices[tx.Id], i)
+		}
+		if bundle.TargetRequired {
+			txHashToBundleIndices[bundle.TargetTxHash] = append(txHashToBundleIndices[bundle.TargetTxHash], i)
+		}
```

Comment:
This is a dangerous overwrite, as it can break logic in cases like b1[tx0, tx1](target: {}) b2[tx2](target: tx1).

Reply:
i is bundles's index. I misunderstood.
Thank you.